### PR TITLE
Gc policy read update

### PR DESCRIPTION
**Pull request by hoangpham95**
Currently when GCPolicy remote resource is updated not on TF (Pantheon, gcloud CLI, etc.), the `gc_rules` string are not updated on read. This PR is fixing that issue.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go [hunk 1]
```diff
@@ -267,6 +276,40 @@ func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) erro
...
+		rules := []interface{}{}
+		for _, c := range gc.(bigtable.IntersectionGCPolicy).Children {
+			rules = append(rules, gcPolicyToGCRuleString(c))
+		}
+		result["rules"] = rules
+	default:
+		break
+	}
+
+	if err := validateNestedPolicy(result, true); err != nil {
```
Comment (hoangpham95) [inline mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go:306]: This is a recursive call so this needs to be considered.

File: mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go [hunk 2]
```diff
@@ -267,6 +276,40 @@ func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) erro
...
 }
 
+func gcPolicyToGCRuleString(gc bigtable.GCPolicy) map[string]interface{} {
+	result := make(map[string]interface{})
+	switch bigtable.GetPolicyType(gc) {
+	case bigtable.PolicyMaxAge:
+		age := gc.(bigtable.MaxAgeGCPolicy).GetDurationString()
+		result["max_age"] = age
+	case bigtable.PolicyMaxVersion:
+		version := gc.(bigtable.MaxVersionsGCPolicy)
```
Comment (hoangpham95) [inline mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go:286]: cast MaxVersionGCPolicy as `float`, not `int`

File: mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go [hunk 3]
```diff
@@ -246,16 +247,24 @@ func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) erro
...
 		return nil
 	}
 
+	log.Printf("[TRACE] Table info: %v, family info length: %v", ti, len(ti.FamilyInfos))
+
 	for _, fi := range ti.FamilyInfos {
-		if fi.Name == name {
+		if fi.Name == columnFamily {
 			d.SetId(fi.GCPolicy)
+			gcRuleJsonString, err := json.Marshal(gcPolicyToGCRuleString(fi.FullGCPolicy))
```
Comment (hoangpham95) [inline mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go:263]: json.Marshal is returning bytes, not string, need to be converted to string

Comment (hoangpham95) [reply mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go:263]: Only set gc_rules when previous field is set too

File: mmv1/third_party/terraform/tests/resource_bigtable_gc_policy_test.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ func TestAccBigtableGCPolicy_basic(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testAccBigtableGCPolicyExists(
 						t, "google_bigtable_gc_policy.policy"),
+						resource.TestCheckResourceAttr("google_bigtable_gc_policy.policy", "gc_rules", ""),
```
Comment (hoangpham95) [inline mmv1/third_party/terraform/tests/resource_bigtable_gc_policy_test.go:33]: Verify not just gc_rules, compare in string if the gc_rules matches with predefined string.

File: mmv1/third_party/terraform/tests/resource_bigtable_gc_policy_test.go [hunk 5]
```diff
@@ -165,6 +165,35 @@ func TestAccBigtableGCPolicy_gcRulesPolicy(t *testing.T) {
...
+		CheckDestroy: testAccCheckBigtableGCPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBigtableGCPolicy_gcRulesCreate(instanceName, tableName, familyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccBigtableGCPolicyExists(
+						t, "google_bigtable_gc_policy.policy"),
+					testAccBigtableCanUpdatePolicy(
+						t, "google_bigtable_gc_policy.policy", bigtable.MaxVersionsPolicy(2)),
+					resource.TestCheckResourceAttr(
```
Comment (hoangpham95) [inline mmv1/third_party/terraform/tests/resource_bigtable_gc_policy_test.go:189]: @kevinsi4508 I haven't found a way to force `Read` before checking this attribute. Does `resource` trigger Read automatically ?

**Review by c2thorn — CHANGES_REQUESTED** [review]
Reviewing now that it has been marked as ready. Couple of things:
Looks like the provider build is failing due to 
```
google-beta/resource_bigtable_gc_policy_test.go:190:44: not enough arguments in call to resource.TestCheckResourceAttr
```
Before pushing your next commit, please read the release note guide and update your release note

**Comment by kevinsi4508** [comment]
This PR is not ready. @hoangpham95 , could you set it to draft mode? Thanks!

**Comment by kevinsi4508** [comment]
To be replaced be https://github.com/GoogleCloudPlatform/magic-modules/pull/6526.

**Pull request closed without merging**